security/advancedtls/advancedtls_test.go

```diff
@@ -47,6 +45,94 @@ func Test(t *testing.T) {
 	grpctest.RunSubTests(t, s{})
 }
 
+// certStore contains all the certificates used in the integration tests.
+type certStore struct {
+	// clientPeer1 is the certificate sent by client to prove its identity.
+	// It is trusted by serverTrust1.
+	clientPeer1 tls.Certificate
+	// clientPeer2 is the certificate sent by client to prove its identity.
+	// It is trusted by serverTrust2.
+	clientPeer2 tls.Certificate
+	// serverPeer1 is the certificate sent by server to prove its identity.
+	// It is trusted by clientTrust1.
+	serverPeer1 tls.Certificate
+	// serverPeer2 is the certificate sent by server to prove its identity.
+	// It is trusted by clientTrust2.
+	serverPeer2 tls.Certificate
+	// serverPeer3 is the certificate sent by server to prove its identity.
+	serverPeer3  tls.Certificate
+	clientTrust1 *x509.CertPool
+	clientTrust2 *x509.CertPool
+	serverTrust1 *x509.CertPool
+	serverTrust2 *x509.CertPool
+}
+
+func readTrustCert(fileName string) (*x509.CertPool, error) {
+	trustData, err := ioutil.ReadFile(fileName)
+	if err != nil {
+		return nil, err
+	}
+	trustBlock, _ := pem.Decode(trustData)
+	if trustBlock == nil {
+		return nil, err
+	}
+	trustCert, err := x509.ParseCertificate(trustBlock.Bytes)
+	if err != nil {
+		return nil, err
+	}
+	trustPool := x509.NewCertPool()
+	trustPool.AddCert(trustCert)
+	return trustPool, nil
+}
+
+// loadCerts function is used to load test certificates at the beginning of
+// each integration test.
+func (cs *certStore) loadCerts() error {
```

Comment:
Optional: None of tests would be able to make any progress when this function returns an error. So, you could consider accepting a `testing.T` as an argument to this function, and directly call `t.Fatal` from here. And also mark this function as a test helper by calling `t.Helper()`.